include/LightGBM/dataset.h

```diff
@@ -328,6 +370,7 @@ class Metadata {
   /*! \brief mutex for threading safe call */
   std::mutex mutex_;
   bool weight_load_from_file_;
+  bool position_load_from_file_;
```

Comment:
Similarly, we should provide a initial value for this in the constructor of `MetaData`.